auth/oauth2adapt/oauth2adapt.go

```diff
@@ -100,8 +100,13 @@ func (ts *tokenSourceAdapter) Token() (*oauth2.Token, error) {
 		Expiry:      tok.Expiry,
 	}
 	// Preserve token metadata.
-	metadata := tok.Metadata
-	if metadata != nil {
+	m := tok.Metadata
+	if m != nil {
+		// Copy map to avoid concurrent map writes error (#11161).
+		metadata := make(map[string]interface{})
```

Comment:
nit: we know at least the current capacity of this map. Maybe we should use that information and allocate +2 incase we add both keys?

Reply:
Fixed, thanks!